spec/labels_checker_spec.rb

```diff
@@ -0,0 +1,173 @@
+# frozen_string_literal: true
+
+require_relative 'spec_helper'
+
+module Danger
+  describe Danger::LabelsChecker do
+    it 'is a plugin' do
+      expect(described_class.new(nil)).to be_a Danger::Plugin
+    end
+
+    describe 'with Dangerfile' do
+      before do
+        @dangerfile = testing_dangerfile
+        @plugin = @dangerfile.labels_checker
+      end
+
+      context 'with required labels' do
+        it 'returns a custom error when a PR does not have at least one label' do
+          allow(@plugin.github).to receive(:pr_labels).and_return([])
+
+          error = 'PR is missing at least one label.'
+          @plugin.check(required_labels: [/.*/], required_labels_error: error)
+
+          expect(@dangerfile.status_report[:warnings]).to be_empty
+          expect(@dangerfile.status_report[:errors]).to eq([error])
+        end
+
+        it 'does nothing when a PR has at least one label' do
+          pr_labels = ['my-label']
+          allow(@plugin.github).to receive(:pr_labels).and_return(pr_labels)
+
+          @plugin.check(required_labels: [/.*/])
+
+          expect(@dangerfile.status_report[:errors]).to be_empty
+          expect(@dangerfile.status_report[:warnings]).to be_empty
+        end
+
+        it 'returns an error containing the required labels when a PR does not meet the label requirements' do
+          pr_label = 'random other label'
+          allow(@plugin.github).to receive(:pr_labels).and_return([pr_label])
+
+          @plugin.check(
+            required_labels: [/feature: .*/]
+          )
```

Comment:
I'd suggest to also test test the case of only some of the labels meeting the requirements but not all.

In fact, none of your test checks the case for multiple `required_labels`, while we need to make sure that:
 - it's not interpreted as "all labels must match one of those RegExes" (while we want to allow `'random other label'` even if it doesn't match, as long as we _also_ have one label like `feature: magic` too that matches)
 - nor being interpreted as "we need one label that matches any of those RegExes" (i.e. if we have `[/feature: .*/, /type: .*/]` then we want a label of each, and just having one label that matches one of the two should not be enough
 - but instead properly interpreted as "we require at least one label matching for each RegEx we provide"

In other words, we need test cases with `required_labels: [/^feature:/, /^type:/]` that would:
 - Fail if `pr_labels = ['random label', 'feature: magic', 'wizard needed']` (ensuring we detect that we're missing a label with `type: .*`)
 - Fail if `pr_labels = ['feature: magic']` (ensuring that we don't accidentally interpret the code as "all labels must match at least one of the `required_labels", but quite the opposite of "all `required_labels` must match at least one label")
 - Pass if `pr_labels = ['random 1', 'feature: magic', 'random 2', 'type: fantasy', 'wizard needed']`, to ensure that not only we check that both `/^feature:/` and `/^type:/` have a match, but also that we're ok if the other labels don't match, as long as one label matched each

Reply:
Thanks a lot! I added more data to cover these scenarios: f717b268ebeea34e6b9409708eea13b801dd3b4b